File: example/new_employee/new_employee.c
CURL *curl;
CURLcode res;

curl = curl_easy_init();

if(curl) {
  curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "POST");
  curl_easy_setopt(curl, CURLOPT_URL, "localhost:3023/uri/rest/sample/tiger/employee");
  curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
  curl_easy_setopt(curl, CURLOPT_DEFAULT_PROTOCOL, "https");

  struct curl_slist *headers = NULL;
  headers = curl_slist_append(headers, "Content-Type: text/plain");

  curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
  const char *data = "!{ name }!\nArmstrong Joe\n\n!{ version }!\n1";
  
  curl_easy_setopt(curl, CURLOPT_POSTFIELDS, data);
  res = curl_easy_perform(curl);
}

curl_easy_cleanup(curl);

File: example/new_employee/new_employee.js
// Using Fetch
//
var myHeaders = new Headers();
myHeaders.append("Content-Type", "text/plain");

var raw = "!{ name }!\nArmstrong Joe\n\n!{ version }!\n1";

var requestOptions = {
  method: 'POST',
  headers: myHeaders,
  body: raw,
  redirect: 'follow'
};

fetch("localhost:3023/uri/rest/sample/tiger/employee", requestOptions)
  .then(response => response.text())
  .then(result => console.log(result))
  .catch(error => console.log('error', error));


// Using jQuery
//
var settings = {
    "url": "localhost:3023/uri/rest/sample/tiger/employee",
    "method": "POST",
    "timeout": 0,
    "headers": {
      "Content-Type": "text/plain"
    },
    "data": "!{ name }!\nArmstrong Joe\n\n!{ version }!\n1",
  };
  
  $.ajax(settings).done(function (response) {
    console.log(response);
  });


// Using XHR
//
var data = "!{ name }!\nArmstrong Joe\n\n!{ version }!\n1";

var xhr = new XMLHttpRequest();
xhr.withCredentials = true;

xhr.addEventListener("readystatechange", function() {
  if(this.readyState === 4) {
    console.log(this.responseText);
  }
});

xhr.open("POST", "localhost:3023/uri/rest/sample/tiger/employee");
xhr.setRequestHeader("Content-Type", "text/plain");

xhr.send(data);